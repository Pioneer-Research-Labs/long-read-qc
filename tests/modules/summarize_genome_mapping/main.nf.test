nextflow_process {

    name "Test Process summarize_genome_mapping"
    script "modules/summarize_genome_mapping/main.nf"
    process "summarize_genome_mapping"

    test("Should generate two files summarizing genome mapping statistics") {

        when {
            params {
                outdir = outputDir
            }
            process {
                """
                // Two input summaries, seq_stats and genome_mapping.
                input[0] = file('${projectDir}/test_data/mapped_genome_map.tsv')
                input[1] = file('${projectDir}/test_data/for_genome_mapping_concatenated_seq_stats.csv')
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
            "$outputDir/concatenated_genome_mapping.csv",
            "$outputDir/genome_mapping_summary.csv"
            ).match()
        }

    }

}
