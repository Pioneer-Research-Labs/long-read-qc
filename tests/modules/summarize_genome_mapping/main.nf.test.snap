{
    "Should generate two files summarizing genome mapping statistics": {
        "content": [
            "concatenated_genome_mapping.csv:md5,2edbfb19e2a087f301fcdb841dee7856",
            "genome_mapping_summary.csv:md5,79a965dd9671179676f220315c06f595"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-15T14:24:00.686337"
    }
}